### PR TITLE
sql: add schema to the `SHOW ENUMS` command

### LINKED ISSUE #52636: sql: `SHOW ENUMS` command should show the schema of the enum as well
Right now, the `SHOW ENUMS` command displays data like this:

```
   name   |                  value
----------+-------------------------------------------
  weekday | monday|tuesday|wednesday|thursday|friday
  weekend | sunday|saturday
```

We should include the schema of the type as well here.

### PR DESCRIPTION
**Pull request by rohany**
Fixes #52636.

This commits adds the schema of the type to the output of the `SHOW
ENUMS` command.

Release note: None

### CODE REVIEW

**Comment by rohany** [comment]
cc @ericharmeling your enums docs will have to change to include this

File: pkg/sql/catalog/catalogkv/catalogkv.go [hunk 1]
```diff
@@ -463,6 +463,22 @@ func MustGetDatabaseDescByID(
 	return desc.(*sqlbase.ImmutableDatabaseDescriptor), nil
 }
 
+// MustGetSchemaDescByID looks up the schema descriptor given its ID,
+// returning an error if the descriptor is not found.
+func MustGetSchemaDescByID(
```
[the remark this answers is not in the document]
Comment (rohany) [reply pkg/sql/catalog/catalogkv/catalogkv.go:468]: yeah theres precedent (`MustGetTableDescByID`, `MustGetDatabaseDescByID`)

File: pkg/sql/delegate/show_enums.go [hunk 2]
```diff
@@ -12,14 +12,27 @@ package delegate
...
+FROM
+	(
+		SELECT
+			nsp.nspname AS schema,
+			type.typname AS name,
+			enum.enumlabel AS label
+		FROM
+			pg_catalog.pg_enum AS enum
+			JOIN pg_catalog.pg_type AS type ON (type.oid = enum.enumtypid)
+			JOIN pg_catalog.pg_namespace AS nsp ON (type.typnamespace = nsp.oid)
```
[the remark this answers is not in the document]
Comment (rohany) [reply pkg/sql/delegate/show_enums.go:28]: `pg_catalog` is always db scoped

**Review by otan — APPROVED** [review]

File: pkg/sql/delegate/show_enums.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (otan) [reply pkg/sql/delegate/show_enums.go:28]: TIL

**Pull request merged**